### ISSUE TITLE
CognitoIdentityCredentials, Access Denied error, not authorized to perform iam:PassRole???

### ISSUE THREAD

**Issue by KMiso** [aws/aws-sdk-js#1909]
Hi,
I am using cognito federated identity ID unauthorized version to set AWS credentials.
I already granted all the permissions for Cognito_myPoolNameUnauth_Role.
And it seems like other services like putting object in s3 bucket or creating dynamoDB table have
no problem.  Everything works well except for creating lambda function code.
I get the same error again and again even after role is set properly  for lambda.

Is this cognito identity credentials problem or iam problem?
I don't understand why iam role will have authorization error of iam:passrole 
with all the permissions given.
Please help me find solution.
Thank you!

error : AccessDeniedException: User: arn:aws:sts::**********:assumed-role/Cognito_myPoolNameUnauth_Role/CognitoIdentityCredentials is not authorized to perform: iam:PassRole on resource: arn:aws:iam::***************:role/lambda_basic_execution

**Comment by AllanFly120** [comment]
Hi @KMiso,
Per [this doc](https://docs.aws.amazon.com/lambda/latest/dg/intro-permission-model.html) says, you may need PassRole permission to enable an IAM role to pass a role to another AWS service, if you haven't done yet. May I ask why do you want to use unauthenticated rule to create lambda function?

**Comment by KMiso** [comment]
I already granted iam:PassRole permisson for unauthorized cognito federated identity user.
I get the same error even when I granted every AWS permission.
I am using unauthenticated user since it is the process when cognito user clicks verification button( cognito user's status is unconfirmed until they click the button)
When user verified their account, I would like to create lambda function for them.
and it has been possible for unauthenticated user to use s3 and dynamodb services
with permission given.

Only creating lambda function is giving me trouble...
Is there anything I am missing?
